File: OasisComputerSystems.API/Models/Ticket.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.ComponentModel.DataAnnotations;

namespace OasisComputerSystems.API.Models
{
    public class Ticket
    {
        public int Id { get; set; }
        [Required]
        public int ClientId { get; set; }
        public Client Client { get; set; }
        [Required]
        public int TicketNo { get; set; }
        [Required]
        public int PriorityId { get; set; }
        public Priority Priority { get; set; }
        [Required]
        public int TicketTypeId { get; set; }
        public TicketType TicketType { get; set; }
        public int? AssignedToId { get; set; }
        public StaffProfile AssignedTo { get; set; }
        [Required]
        public int SystemModuleId { get; set; }
        public SystemModule SystemModule { get; set; }
        [Required]
        [StringLength(255)]
        public string Subject { get; set; }
        [Required]
        public string ProblemDescription { get; set; }
        public int SubmittedById { get; set; }
        public StaffProfile SubmittedBy { get; set; }
        public DateTime SubmittedOn { get; set; }

        public int? ClosedById { get; set; }
        public StaffProfile ClosedBy { get; set; }
        public DateTime? ClosedOn { get; set; }

        public int? ApprovedById { get; set; }
        public StaffProfile ApprovedBy { get; set; }
        public DateTime? ApprovedOn { get; set; }

        public ICollection<TicketNote> TicketNotes { get; set; }

        public Ticket()
        {
            TicketNotes = new Collection<TicketNote>();
        }

    }
}

File: OasisComputerSystems-SPA/src/app/_models/ticket.ts
import { Client } from './client';

export interface Ticket {
  id: number;
  clientId: number;
  Client?: Client;
  TicketNo?: number;
  priorityId: number;
  priority?: string;
  ticketTypeId: number;
  ticketType?: string;
  assignedToId: number;
  assignedTo?: string;
  systemModuleId?: number;
  systemModule?: string;
  subject?: string;
  problemDescription?: string;
  submittedById?: number;
  submittedBy?: string;
  submittedOn?: Date;
  closedById?: number;
  closedBy?: string;
  closedOn?: Date;
  approvedById?: number;
  approvedBy?: string;
  approvedOn?: Date;
  ticketNotes?: TicketNote;
}

export interface TicketNote {
  id: number;
  ticketId: number;
  notes: string;
  oasisComment: boolean;
  createdById: number;
  createdBy: string;
  createdOn: Date;
}


